### PR TITLE
Infinity Gauntlet Test Merge

### PR DESCRIPTION
**Pull request by sam0oneau**
Adds an Infinity Gauntlet to the game for admin's use in events.

It is designed to be a test merge and can be removed after the event.

#### Changelog
Adds 6 Infinity gems.
Adds 6 Gem artifacts.
Adds an Infinity gauntlet
Adds spells specific to the gauntlet and artifacts.

### CODE REVIEW

**Comment by AsV9** [comment]
what could possibly go wrong

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+
+/obj/item/clothing/gloves/gauntlet/attackby(obj/item/I, mob/user, params)
+	var/obj/item/infinity_stone/S = I
+	if(istype(I, /obj/item/infinity_stone))
+		add_stone(S.gem_type, user)
+		qdel(I)
+		user.dropItemToGround(src)
+		to_chat(user, "<span class='warning'>The gauntlet becomes too hot to touch.</span>")
+
+////obj/item/t_scanner/attack_self(mob/user)
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:126]: ?

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+
+/obj/item/clothing/gloves/gauntlet/proc/snap(user)
+	if(applied_gems.len == 6)
+		to_chat(user, "<span class='warning'>You destroy humanity</span>")
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:130]:
```suggestion
		to_chat(user, "<span class='warning'>You bring balance to humanity</span>")
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+		for(var/mob/living/carbon/M in GLOB.mob_list)
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:131]:
```suggestion
		for(var/mob/living/carbon/M in GLOB.alive_mob_list)
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+			if(M.stat != DEAD)
+				if(!(user == M))
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:133]:
```suggestion
				if(!(user == M) && prob(50))
```
You know, don't kill all of em innit

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+					M.gib()
+	else
+		to_chat(user, "<span class='warning'>You need all the infinity gems to destroy humanity.</span>")
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:136]:
```suggestion
		to_chat(user, "<span class='warning'>You need all the infinity gems to bring balance to humanity.</span>")
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
+
+/obj/item/clothing/gloves/gauntlet/proc/remove_stone(Old_stone) //Not used right now and not fully implemented
+	cut_overlays()
+	for(var/N in applied_gems)
+		var/image/O = image(icon = 'yogstation/icons/obj/gauntlet.dmi', icon_state = "stone_[N]")
+		add_overlay(O)
+
+//Spells
+
+/obj/effect/proc_holder/spell/targeted/tesla/gauntlet
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:193]: a bunch of these spells have variables that are identical to the ones they inherit, like the name and description, so they don't need to be here as well.

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 7]
```diff
@@ -0,0 +1,611 @@
...
+	icon = 'yogstation/icons/obj/gauntlet.dmi'
+	action_icon = 'yogstation/icons/obj/gauntlet.dmi'
+	base_icon_state = "gem_4"
+	action_icon_state = "gem_4"
+	active_icon_state = "gem_4"
+	active_msg = "You prepare to throw your healing bolt!"
+	deactive_msg = "You absorb your healing bolt... for now."
+	active = FALSE
+
+/obj/effect/proc_holder/spell/aimed/resurrect/update_icon()
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:243]: Should probably be finished

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 8]
```diff
@@ -0,0 +1,611 @@
...
+	icon = 'yogstation/icons/obj/gauntlet.dmi'
+	action_icon = 'yogstation/icons/obj/gauntlet.dmi'
+	base_icon_state = "gem_5"
+	action_icon_state = "gem_5"
+	active_icon_state = "gem_5"
+	active_msg = "You prepare to throw your time bolt!"
+	deactive_msg = "You absorb your time bolt... for now."
+	active = FALSE
+
+/obj/effect/proc_holder/spell/aimed/time/update_icon()
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:263]: see above

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 9]
```diff
@@ -0,0 +1,611 @@
...
+	icon = 'yogstation/icons/obj/gauntlet.dmi'
+	action_icon = 'yogstation/icons/obj/gauntlet.dmi'
+	base_icon_state = "gem_6"
+	action_icon_state = "gem_6"
+	active_icon_state = "gem_6"
+	active_msg = "You prepare to throw your healing bolt!"
+	deactive_msg = "You absorb your healing bolt... for now."
+	active = FALSE
+
+/obj/effect/proc_holder/spell/aimed/animate/update_icon()
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:291]: see above

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 10]
```diff
@@ -0,0 +1,611 @@
...
+	icon_state = "artifact_1"
+	item_state = "artifact_1"
+	delay = 3
+	ammo_type = list(/obj/item/ammo_casing/energy/beam_rifle/hitscan/artifact)
+	cell_type = /obj/item/stock_parts/cell/infinite
+	aiming_time = 3
+	recoil = 0
+	pin = /obj/item/firing_pin
+
+/obj/item/gun/energy/beam_rifle/power_artifact/update_icon()
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:309]: see above

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 11]
```diff
@@ -0,0 +1,611 @@
...
+	//Removes the overlays
+
+/obj/item/gun/energy/beam_rifle/power_artifact/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/gloves/gauntlet))
+		to_chat(user, "You destroy the artifact with the gauntlet, leaving just an infinity stone.")
+		qdel(src)
+
+/obj/item/gun/energy/beam_rifle/power_artifact/Destroy()
+	new /obj/item/infinity_stone/power(loc, 1)
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:318]:
```suggestion
	new /obj/item/infinity_stone/power(get_turf(src), 1)
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 12]
```diff
@@ -0,0 +1,611 @@
...
+
+/obj/effect/projectile/tracer/tracer/beam_rifle/artifact
+	name = "laser"
+	icon_state = "hcult"
+
+//Space stone artifact
+
+/obj/item/teleportation_scroll/space_artifact
+	name = "The Tesseract"
+	desc = "A cube of immense power. It's teleportation powers are unimaginable."
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:335]:
```suggestion
	desc = "A cube of immense power. Its teleportation powers are unimaginable."
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 13]
```diff
@@ -0,0 +1,611 @@
...
+	icon = 'icons/obj/wizard.dmi'
+	icon = 'yogstation/icons/obj/gauntlet.dmi'
+	icon_state = "artifact_2"
+	uses = 100000 //A number so large it'll never be reached
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:339]:
```suggestion
	uses = INFINITY //A number so large it'll never be reached
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 14]
```diff
@@ -0,0 +1,611 @@
...
+	onclose(user, "scroll")
+	return
+
+/obj/item/teleportation_scroll/space_artifact/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/gloves/gauntlet))
+		to_chat(user, "You destroy the artifact with the gauntlet, leaving just an infinity stone.")
+		qdel(src)
+
+/obj/item/teleportation_scroll/space_artifact/Destroy()
+	new /obj/item/infinity_stone/space(loc, 1)
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:355]:
```suggestion
	new /obj/item/infinity_stone/space(get_turf(src), 1)
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 15]
```diff
@@ -0,0 +1,611 @@
...
+	item_state = "artifact_3"
+	max_charges = 1000000
+
+/obj/item/gun/magic/staff/reality_artifact/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/gloves/gauntlet))
+		to_chat(user, "You destroy the artifact with the gauntlet, leaving just an infinity stone.")
+		qdel(src)
+
+/obj/item/gun/magic/staff/reality_artifact/Destroy()
+	new /obj/item/infinity_stone/reality(loc, 1)
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:375]:
```suggestion
	new /obj/item/infinity_stone/reality(get_turf(src), 1)
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 16]
```diff
@@ -0,0 +1,611 @@
...
+	projectile_type = /obj/item/projectile/magic/reality
+
+/obj/item/projectile/magic/reality
+	name = "bolt of reality"
+	icon_state = "red_1"
+	damage = 0
+	damage_type = BURN
+	nodamage = TRUE
+
+	////obj/item/t_scanner/attack_self(mob/user)
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:388]: ???

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 17]
```diff
@@ -0,0 +1,611 @@
...
+		else
+			to_chat(user, "<span class='warning'>[C] doesn't have two hands...</span>")
+
+/obj/item/restraints/handcuffs/soul_artifact/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/gloves/gauntlet))
+		to_chat(user, "You destroy the artifact with the gauntlet, leaving just an infinity stone.")
+		qdel(src)
+
+/obj/item/restraints/handcuffs/soul_artifact/Destroy()
+	new /obj/item/infinity_stone/soul(loc, 1)
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:463]:
```suggestion
	new /obj/item/infinity_stone/soul(get_turf(src), 1)
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 18]
```diff
@@ -0,0 +1,611 @@
...
+/obj/item/soul_essence/Initialize()
+	. = ..()
+	addtimer(CALLBACK(src, .proc/go_inert_soul),300)
+
+/obj/item/soul_essence/proc/go_inert_soul()
+	name = "regenerative essence"
+	desc = "An empty essence."
+	icon = 'yogstation/icons/obj/gauntlet.dmi'
+	icon_state = "soul_essence_empty"
+	empty = 1
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:485]: I'm guessing this needs to be a TRUE

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 19]
```diff
@@ -0,0 +1,611 @@
...
+/obj/item/soul_essence/afterattack(atom/target, mob/user, proximity_flag)
+	. = ..()
+	if(proximity_flag && ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(empty)
+			to_chat(user, "<span class='notice'>[src] is empty and can no longer be used to heal.</span>")
+			return
+		else
+			if(H.stat == DEAD)
+				to_chat(user, "<span class='notice'>[src] are useless on the dead.</span>")
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:496]:
```suggestion
				to_chat(user, "<span class='notice'>[src] is useless on the dead.</span>")
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 20]
```diff
@@ -0,0 +1,611 @@
...
+	var/toggle = 0
+
+/obj/item/clothing/neck/time_artifact/equipped(mob/user, slot)
+	..()
+	var/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop/J = new /obj/effect/proc_holder/spell/aoe_turf/conjure/timestop()
+	timestop_definitions(J)
+	if(!toggle)
+		user.AddSpell(J)
+		user.add_movespeed_modifier(type, update=TRUE, priority=100, multiplicative_slowdown=-2, blacklisted_movetypes=(FLYING|FLOATING))
+	toggle = 1
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:523]:
```suggestion
	toggle = TRUE
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 21]
```diff
@@ -0,0 +1,611 @@
...
+
+/obj/item/clothing/neck/time_artifact/dropped(mob/user)
+	..()
+	user.remove_movespeed_modifier(type)
+	user.RemoveSpell(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop)
+	toggle = 0
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:529]:
```suggestion
	toggle = FALSE
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 22]
```diff
@@ -0,0 +1,611 @@
...
+
+/obj/item/clothing/neck/time_artifact/afterattack(atom/target, mob/user, proximity)
+	if(istype(target, /obj/machinery/hydroponics))
+		to_chat(user, "<span class='warning'>The plants begin to grow rapidly! Soon they will be ready to harvest.</span>")
+		plant_growth(target)
+
+/obj/item/clothing/neck/time_artifact/proc/plant_growth(obj/machinery/hydroponics/H)
+	H.age += 10
+
+/proc/timestop_definitions(obj/effect/proc_holder/spell/aoe_turf/conjure/timestop/S) //Timestop needs to have the same path to ensure its host can timestop immune.
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:539]: Why is this a global proc?

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 23]
```diff
@@ -0,0 +1,611 @@
...
+	S.clothes_req = FALSE
+
+/obj/item/clothing/neck/time_artifact/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/gloves/gauntlet))
+		to_chat(user, "You destroy the artifact with the gauntlet, leaving just an infinity stone.")
+		qdel(src)
+
+/obj/item/clothing/neck/time_artifact/Destroy()
+	new /obj/item/infinity_stone/time(loc, 1)
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:548]:
```suggestion
	new /obj/item/infinity_stone/time(get_turf(src), 1)
```

File: yogstation/code/modules/gauntlet/gauntlet.dm [hunk 24]
```diff
@@ -0,0 +1,611 @@
...
+			to_chat(user, "<span class='notice'>[SM] looks interested for a moment, but then looks back down. Maybe you should try again later.</span>")
+			..()
+
+/obj/item/mind/mind_artifact/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/gloves/gauntlet))
+		to_chat(user, "You destroy the artifact with the gauntlet, leaving just an infinity stone.")
+		qdel(src)
+
+/obj/item/mind/mind_artifact/Destroy()
+	new /obj/item/infinity_stone/mind(loc, 1)
```
Comment (Amelia0010) [inline yogstation/code/modules/gauntlet/gauntlet.dm:594]:
```suggestion
	new /obj/item/infinity_stone/mind(get_turf(src), 1)
```

**Review by Amelia0010 — CHANGES_REQUESTED** [review]

**Comment by SomeguyManperson** [comment]
is this kill?

**Comment by sam0oneau** [comment]
Indeed

**Comment by Amelia0010** [comment]
mane then close the PR

**Pull request closed without merging**